src/libPOW/pow.cpp

```diff
@@ -433,19 +435,24 @@ void POW::InitCUDA() {
 #ifdef CUDA_MINE
   using namespace dev::eth;
 
+  auto gpuToUse = GetGpuToUse();
+  auto deviceGenerateDag = *gpuToUse.begin();
+  LOG_GENERAL(INFO, "Generate dag Nvidia GPU #" << deviceGenerateDag);
+
   if (!CUDAMiner::configureGPU(CUDA_BLOCK_SIZE, CUDA_GRID_SIZE, CUDA_STREAM_NUM,
-                               CUDA_SCHEDULE_FLAG, 0, 0, false, false)) {
+                               CUDA_SCHEDULE_FLAG, 0, deviceGenerateDag, false,
+                               false)) {
     LOG_GENERAL(FATAL, "Failed to configure CUDA GPU, please check hardware");
   }
 
-  CUDAMiner::setNumInstances(UINT_MAX);
-  auto gpuToUse = GetGpuToUse();
+  CUDAMiner::setNumInstances(gpuToUse.size());
+
   auto totalGpuDevice = CUDAMiner::getNumDevices();
   for (const auto gpuIndex : gpuToUse) {
     if (gpuIndex >= totalGpuDevice) {
       LOG_GENERAL(FATAL, "Selected GPU "
                              << gpuIndex
-                             << " exceed the physical OpenCL GPU number "
+                             << " exceed the physical Nvidia GPU number "
```

Comment:
Not sure if it's proper to have other company's name inside our source code.

Reply:
I think should be ok. If you search Nvidia in all github, you can find 15 Million matchs.